src/prefect/utilities/filesystem.py

```diff
@@ -123,5 +123,12 @@ def relative_path_to_current_platform(path_str: str) -> Path:
 
 
 def get_open_file_limit():
-    soft_limit, _ = resource.getrlimit(resource.RLIMIT_NOFILE)
+    """Get the open file count limit"""
+    if platform.system() != 'Windows':
+        import resource
+        soft_limit, _ = resource.getrlimit(resource.RLIMIT_NOFILE)
+    else:
+        import win32file
+        soft_limit = win32file._getmaxstdio()
```

Comment:
I think we can use a safe default value when not on Linux or MacOS so that we don't need to introduce a new dependency.
```suggestion
    """Get the open file count limit"""
    if sys.platform == 'linux' or sys.platform == 'darwin':
        import resource
        soft_limit, _ = resource.getrlimit(resource.RLIMIT_NOFILE)
    else:
        soft_limit = 200
```

Reply:
@desertaxle You can also call the same `getmaxstdio` function in the Windows C runtime without adding an import. Draft PR here for comparison, with a test added as well: https://github.com/PrefectHQ/prefect/pull/10059

Hardcoding a default value is simpler, though, however Windows will almost always return a value higher than 200.